### PR TITLE
Feature: Introduce some flags on notary verify

### PR DESCRIPTION
**Pull request by HuKeping**
There is no reason to give out all the payload especially when it is
huge engough, it will definitely ruin the console.

Signed-off-by: Hu Keping <hukeping@huawei.com>

### CODE REVIEW

**Comment by riyazdf** [comment]
I'm not sure if we'd like a drastic change in behavior, since this change would break this use case:
`curl <binary> | notary verify | sh`

How about adding a short-form mode instead?  We could even have the short-form mode only return success or failure via error codes, and not even print any output.

**Comment by endophage** [comment]
I think the desired functionality here probably deserves a whole set of flags on `notary verify` focussed on verifying a file already on disk, like `--input`, `--output`, and `--quiet` (`-i`, `-o`, and `-q` for short forms) which would make it align with a lot of other *nix tools that use stdin/stdout by default and take flags for other I/O options.

For interaction between `--output` and `--quiet` I'd expect either an error saying both flags were provided and only one can be used. Or `--quiet` takes priority and `--output` is ignored.

**Comment by HuKeping** [comment]
Hmm,  a set of flags is good for that, will change.

**Comment by cyli** [comment]
jenkins, test this please

**Comment by endophage** [comment]
Love it. LGTM! Thank you!

File: cmd/notary/util.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	// Flag "quiet" was not "true", that's why we get here.
+	if t.output != "" {
+		if err := ioutil.WriteFile(t.output, payload, 0644); err != nil {
+			return err
+		}
+	}
+
+	os.Stdout.Write(payload)
```
Comment (riyazdf) [inline cmd/notary/util.go]: it seems that if we provide a value to the `output` flag, we write both to the file as well as stdout.
To be consistent with the flag description, we can remove the error-checking in the `t.output != ""` case and just `return ioutil.WriteFile(t.output, payload, 0644)`.

I'm also ok with leaving this behavior of writing to both the output file and stdout, if we update the flag description

File: cmd/notary/util.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"os"
+)
+
+// getPayload is a helper function to get the content used to be verified
+// either from an existing file or STDIN.
+func getPayload(t *tufCommander) ([]byte, error) {
+
+	// Reads from the given file
+	if t.input != "" {
+		// Please be noticed that ReadFile will cut off the size if it was over 1e9.
```
Comment (riyazdf) [inline cmd/notary/util.go]: nit: `Please note that`

File: cmd/notary/util.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("Error reading content from STDIN: %v", err)
+	}
+	return payload, nil
+}
+
+// feedback is a helper function to print the payload to a file or STDOUT or keep quiet
+// due to the value of flag "quiet" and "output".
+func feedback(t *tufCommander, payload []byte) error {
+	// We only get here when everythings goes well, since the flag "quiet" was
```
Comment (riyazdf) [inline cmd/notary/util.go]: nit typo: `everything`

File: cmd/notary/util.go (hunk 1 above)
Comment (endophage) [inline cmd/notary/util.go]: Good catch, I totally missed that! Yeah, we should only write to the file. That will be the expected behaviour for anyone familiar with *nix

Comment (HuKeping) [inline cmd/notary/util.go]: Thanks for noticing that! Will change.

**Comment by HuKeping** [comment]
Updated

**Comment by riyazdf** [comment]
Jenkins, test this please.

**Comment by riyazdf** [comment]
thanks for the update, this is a great contribution!  LGTM!

**Pull request merged**